ts/main.ts

```diff
@@ -147,6 +147,100 @@ if (!$monsterPage) throw new Error('Could not load Monster Page');
+  $favTitle?.classList.add('hidden');
+  $monTitle?.classList.remove('hidden');
+  $filter.classList.remove('hidden');
+  $star.classList.add('hidden');
+  $star?.classList.remove('favorite');
+});
+
+if (!$estrella) throw new Error('could not load favorite star click');
+$estrella.addEventListener('click', function () {
+  if ($star.classList[1] === 'favorite') {
+    removeFav();
+  } else if ($star.classList[1] !== 'favorite') {
+    favMon();
+  }
+});
+
+function removeFav(): void {
+  const $card = document.querySelector('.monster-card');
+  if (!$nameMon) throw new Error('Could not load name');
+  if (!$card) throw new Error('Could not load name');
+  if (!$monFavCon?.firstChild)
+    throw new Error('could not load the first child element');
+  if (!$nameMon.textContent) throw new Error('Could not load name');
+  $star?.classList.remove('favorite');
+  for (let i = 0; i < data.favorite.length; i++) {
+    if (Number($nameMon.id) === data.favorite[i].id) {
+      data.favorite.splice(i, 1);
+    }
+  }
+  writeData();
+  if (!$monFav) throw new Error('could not load monster fav page');
+  const $cardToRemove = $monFav.querySelector(
+    `[data-monster-id='${Number($nameMon.id)}']`,
+  );
+  $cardToRemove?.remove();
+}
+
+function favMon(): void {
+  if (!$nameMon) throw new Error('Could not load name');
+  if (!$nameMon.textContent) throw new Error('Could not load name');
+  for (let i = 0; i < data.favorite.length; i++) {
+    if ($nameMon.textContent === data.favorite[i].name) {
+      return;
+    }
+  }
+  $star?.classList.add('favorite');
+  const favMon = {
+    name: $nameMon.textContent,
+    photoURL: `images/icons/${$nameMon.textContent}.webp`,
+    id: Number($nameMon.id),
+  };
+  data.favorite.push(favMon);
+  $monFavCon?.append(renderMonster(favMon));
+  writeData();
+  readData();
+}
+
+if (!$favoriteTab) throw new Error('Could not load Monster Page');
+$favoriteTab.addEventListener('click', function () {
+  $monLine.classList.add('hidden');
+  $monSelect.classList.add('hidden');
+  $monFav.classList.remove('hidden');
+  $favoriteTab?.classList.add('selected');
+  $monsterPage.classList.remove('selected');
+  $favTitle?.classList.remove('hidden');
+  $monTitle?.classList.add('hidden');
   $filter.classList.remove('hidden');
   $star.classList.add('hidden');
+  $star?.classList.remove('favorite');
 });
+
+document.addEventListener('DOMContentLoaded', domTree);
+function domTree(): void {
+  async function makeMonsters(): Promise<void> {
+    if (!$monChoice) throw new Error('Could not load row');
+    try {
+      const response = await fetch(`https://mhw-db.com/monsters`);
+      if (!response.ok) {
+        throw new Error('Could not load the response');
+      }
+      const monster = await response.json();
+      for (let i = 0; monster.length + 1; i++) {
+        const newPage = renderMonster(monster[i]);
+        $monChoice.append(newPage);
+      }
+    } catch (error) {}
+  }
+  makeMonsters();
```

Comment:
For this we just needed to move the `makeMonsters` function call into the `domTree` function